lib/push.py

```diff
@@ -82,19 +84,51 @@ def manifest_new(class_):
 
    def cleanup(self):
       ch.INFO("cleaning up")
-      # Delete the tarballs since we can't yet cache them.
-      for (_, tar_c) in self.layers:
-         ch.VERBOSE("deleting tarball: %s" % tar_c)
-         tar_c.unlink_()
-
-   def prepare(self):
-      """Prepare self.image for pushing to self.dst_ref. Return tuple: (list
-         of gzipped layer tarball paths, config as a sequence of bytes,
-         manifest as a sequence of bytes).
-
-         There is not currently any support for re-using any previously
-         prepared files already in the upload cache, because we don't yet have
-         a way to know if these have changed until they are already build."""
+     # # Delete the tarballs since we can't yet cache them.
+     # for (_, tar_c) in self.layers:
+     #    ch.VERBOSE("deleting tarball: %s" % tar_c)
+     #    tar_c.unlink_()
+
+   def get_prev_prepared(self):
+      """Check self.image if an existing config, manifest, and tar exist.
+         Return list: [config, manifest, tar] if these files exist or None
+         if not."""
+      (sid, git_hash) = bu.cache.find_image(self.image)
+
+      conf_suffix = git_hash + ".config.json"
+      conf_path = ch.storage.upload_cache // conf_suffix
+      try:
+         json.load(open(conf_path))
+      except:
+         print("Previous config does not exist")
+         return None
```

Comment:
Try to avoid using try catch blocks when possible. Here we can check if the file exists before attempting to load the json contents. See the module `os`; I believe there are function wrappers in one of the charliecloud library that wraps it, e.g., `ossafe()`.

Also, if there is an existing function that loads json file contents use that, rather than `json.load()`.